tests/gold_tests/pluginTest/test_hooks/body_buffer.test.py

```diff
@@ -0,0 +1,129 @@
+                           "body": self.encoded_chunked_request}
+        self.chunked_response_body = "chunked response"
+        self.encoded_chunked_response = "{0}\r\n{1}\r\n0\r\n\r\n".format(
+                int_to_hex_string(len(self.chunked_response_body)), self.chunked_response_body)
+        response_header2 = {"headers": "HTTP/1.1 200 OK\r\n"
+                            "Transfer-Encoding: chunked\r\n"
+                            "Server: microserver\r\n"
+                            "Connection: close\r\n\r\n",
+                            "timestamp": "1469733493.993",
+                            "body": self.encoded_chunked_response}
+        self._server.addResponse("sessionlog.json", request_header2, response_header2)
+
+    def setupTS(self):
+        self._ts = Test.MakeATSProcess("ts", select_ports=False)
+        self._ts.Disk.remap_config.AddLine(
+            'map / http://127.0.0.1:{0}'.format(self._server.Variables.Port)
+        )
+        Test.PreparePlugin(os.path.join(Test.Variables.AtsTestToolsDir, 'plugins', 'request_buffer.c'), self._ts)
+        self._ts.Disk.records_config.update({
+            'proxy.config.diags.debug.enabled': 1,
+            'proxy.config.diags.debug.tags': 'request_buffer',
+            'proxy.config.http.per_server.connection.max': self._origin_max_connections,
+            # Disable queueing when connection reaches limit
+            'proxy.config.http.per_server.connection.queue_size': 0,
+        })
+
+        self._ts.Streams.stderr = Testers.ContainsExpression(
+            "request_buffer_plugin gets the request body with length\[{0}\]: \[{1}\]".format(
+                len(self.content_length_request_body), self.content_length_request_body),
+            "Verify that the plugin parsed the content-length request body data.")
+        self._ts.Streams.stderr += Testers.ContainsExpression(
+            "request_buffer_plugin gets the request body with length\[{0}\]: \[".format(
+                len(self.encoded_chunked_request)),
+            "Verify that the plugin parsed the chunked request body.")
+        self._ts.Streams.stderr += Testers.ContainsExpression(
+            "^{}".format(self.chunked_request_body),
+            "Verify that the plugin parsed the chunked request body data.")
+
+    def run(self):
+        tr = Test.AddTestRun()
+        # Send both a Content-Length request and a chunked-encoded request.
+        tr.Processes.Default.Command = (
+                'curl -v http://127.0.0.1:{0}/contentlength -d "{1}" ; '
+                'curl -v http://127.0.0.1:{0}/chunked -H "Transfer-Encoding: chunked" -d "{2}"'.format(
+                    self._ts.Variables.port, self.content_length_request_body, self.chunked_request_body))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.StartBefore(self._server)
+        tr.Processes.Default.StartBefore(Test.Processes.ts)
+        tr.Processes.Default.Streams.stderr = "200.gold"
+
+
```

Comment:
Extra blank lines are dangerous as @zwoop hates them and his forefathers invented berserk.

Reply:
I try to format my Python files according to the PEP 8 standard. In particular:

https://www.python.org/dev/peps/pep-0008/#blank-lines

> Surround top-level function and class definitions with two blank lines.

As an aside, in case it's helpful to others, I use the flake8 utility via the Ale VIM plugin to help me accomplish PEP 8 conformance. (I think this requires VIM 8.)

Reply:
Sounds good zwoop's canceling your Blood Eagle.